bigquery_etl/view/generate_stable_views.py

```diff
@@ -241,6 +217,25 @@ def get_stable_table_schemas() -> List[SchemaFile]:
     ]
 
 
+def schemas_uri_for_project(target_project):
+    """Return URI for the schemas tarball deployed to the given project.
+
+    We construct a fake query and send it to the dry run service in order
+    to read dataset labels, which contains the commit hash associated
+    with the most recent production schemas deploy.
+    """
+    with tempfile.TemporaryDirectory() as tdir:
+        d = Path(tdir) / target_project / "telemetry_derived" / "myquery"
+        d.mkdir(parents=True)
+        tfile = Path(d) / "query.sql"
+        with tfile.open("w") as f:
+            f.write("SELECT 1")
+        dryrun = DryRun(str(tfile))
+        build_id = dryrun.get_dataset_labels()["schemas_build_id"]
```

Comment:
`DryRun` does not support checking the project directory, and the service does not allow the default dataset to include a project, so accepting `target_project` here is misleading.

Also, if `content` is passed to `DryRun`, then `sqlfile` doesn't need to exist.

```suggestion
def schemas_uri():
    """Return URI for the schemas tarball deployed to moz-fx-data-shared-prod.

    We construct a fake query and send it to the dry run service in order
    to read dataset labels, which contains the commit hash associated
    with the most recent production schemas deploy.
    """
    dryrun = DryRun(
        sqlfile=str(Path("telemetry_derived") / "table" / "query.sql"),
        content="SELECT 1",
    )
    build_id = dryrun.get_dataset_labels()["schemas_build_id"]
```

Reply:
> Also, if content is passed to DryRun, then sqlfile doesn't need to exist.

This makes me so happy!